### ISSUE TITLE
CADL APIView filters out things in the global namespace

### ISSUE THREAD

**Issue by catalinaperalta** [Azure/azure-sdk-tools#5289]
Can we add the CADL DPG modifications that are defined in client.cadl? 

Example CADL that did not include client.cadl definitions: https://apiview.dev/Assemblies/Review/205e1a393e0c4ed8b5a42fb5ba5988cd

PR for the CADL API view shared above: https://github.com/Azure/azure-rest-api-specs-pr/pull/10411

**Comment by catalinaperalta** [comment]
cc @maririos @praveenkuttappan

**Comment by tjprescott** [comment]
@catalinaperalta you can already do this by generating a token file manually (likely not by using a URL). 

Since this pertains to the automatically generated APIView, this would fall to Central EngSys.

**Comment by maririos** [comment]
@tjprescott you mean that the automatically generated APIView from that PR is not correct? what it is missing?

**Comment by tjprescott** [comment]
@maririos the automatic PR is only targeting main.cadl. @catalinaperalta wants it to target client.cadl (which will pull in main.cadl) so that she sees the effect of the sidecar file.

However... for automatic PRs, it doesn't really make sense to apply the sidecar. It's supposed to a be a view of the service, not the client. So perhaps in this use case, the only appropriate course of action _is_ to make a manual APIView targeting the client.cadl. In theory this could be done with the URL-based manual review as well by pointing to the client.cadl file.

**Comment by tjprescott** [comment]
So assuming that the manual PR where you target a URL works by targeting a client.cadl file, I think this issue could actually be closed.

**Comment by catalinaperalta** [comment]
@tjprescott I manually generated to token file for the CADL APIview and uploaded that one. I did not use the URL since I wasnt sure what URL was expected. So should the manual token file have showcased the client.cadl? I did not see it in the API view I uploaded.

**Comment by catalinaperalta** [comment]
Maybe I'm on an older version of the package? This is what I'm using:

```
"@azure-tools/cadl-apiview": "^0.3.4",
```

**Comment by tjprescott** [comment]
The manual token file (assuming you targeted client.cadl) should include all of your augment decorators.

**Comment by maririos** [comment]
@praveenkuttappan could you help with this?
> @tjprescott I manually generated to token file for the CADL APIview and uploaded that one. I did not use the URL since I wasnt sure what URL was expected. So should the manual token file have showcased the client.cadl? I did not see it in the API view I uploaded.

More on issue: https://github.com/Azure/azure-sdk-tools/issues/5273

**Comment by tjprescott** [comment]
@praveenkuttappan targeting a client.cadl file by running the parser locally already works. The question is whether it works for the URL-based manual APIView.

**Comment by tjprescott** [comment]
> @tjprescott I manually generated to token file for the CADL APIview and uploaded that one. I did not use the URL since I wasnt sure what URL was expected. So should the manual token file have showcased the client.cadl? I did not see it in the API view I uploaded.

@catalinaperalta what is the link to the APIView generated and what command did you use to generate it? It should look something like: `cadl compile client.cadl --emit=@azure-tools/cadl-apiview`.

**Comment by praveenkuttappan** [comment]
> @praveenkuttappan targeting a client.cadl file by running the parser locally already works. The question is whether it works for the URL-based manual APIView.
> 
> Even though the issue requested it, I don't think it makes any sense for automatic APIViews to target client.cadl (or to even enable that).

So I assume we don't need any change in central engsys side. We have another issue to track improving guidelines on how to create API review for CADL.

**Comment by catalinaperalta** [comment]
Sorry for the delay @tjprescott, I did use exactly that command. I also just tried it again with the same result. 

Command: cadl compile .\client.cadl --emit="@azure-tools/cadl-apiview"
APIview: https://apiview.dev/Assemblies/Review/3d80f1b45fec4bc4b97a39c991948d79

**Comment by catalinaperalta** [comment]
I see. Is the namespace required for the CADL Azure DPG declarations? If so, I think we may need to document this somewhere and/or have it as a linter rule.

**Comment by tjprescott** [comment]
Well, if you put it any namespace starting with `HealthDecisionsApi` it will render. That is the APIView logic because otherwise it will render literally everything under the sun from Cadl, Azure.Core, etc. However the logic is flawed because it is supposed to support your package namespace and subnamespaces, but in this case you can have a package namespace of `Foo` and it will happily accept `Foo2` which isn't a subnamespace at all. So there is some logic for me to revisit (especially since it relies on now-deprecated stuff). Whether to render things in the global namespace... I'm not sure.

Also, I opened this issue (https://github.com/microsoft/cadl/issues/1625) that I ran into while looking into this. But for now you should be unblocked.
